src/contribute/release-guide.md

```diff
@@ -178,64 +231,78 @@ Javadoc to the Javadoc for other modules that Beam depends on.
 
 Check out the version of the codebase from which you start the release. For a new minor or major release, this may be `HEAD` of the `master` branch. To build a hotfix/incremental release, instead of the `master` branch, use the release tag of the release being patched. (Please make sure your cloned repository is up-to-date before starting.)
 
     git checkout <master branch OR release tag>
 
+**NOTE**: If you are doing an incremental/hotfix release (e.g. 2.5.1), please check out the previous release tag, rather than the master branch.
 
 Set up a few environment variables to simplify Maven commands that follow. (We use `bash` Unix syntax in this guide.)
 
-    VERSION="1.2.3"
-    NEXT_VERSION="1.2.4"
-    BRANCH_NAME="release-${VERSION}"
-    DEVELOPMENT_VERSION="${NEXT_VERSION}-SNAPSHOT"
+    RELEASE=2.5.0
+    NEXT_VERSION_IN_BASE_BRANCH=2.6.0
+    BRANCH=release-${RELEASE}
 
 Version represents the release currently underway, while next version specifies the anticipated next version to be released from that branch. Normally, 1.2.0 is followed by 1.3.0, while 1.2.3 is followed by 1.2.4.
 
-Use Maven release plugin to create the release branch and update the current branch to use the new development version. This command applies for the new major or minor version. (Warning: this command automatically pushes changes to the code repository.)
+**NOTE**: Only if you are doing an incremental/hotfix release (e.g. 2.5.1), please check out the previous release tag, before running the following instructions:
+
+    BASE_RELEASE=2.5.0
+    RELEASE=2.5.1
+    NEXT_VERSION_IN_BASE_BRANCH=2.5.0
+    git checkout tags/${BASE_RELEASE}
+
+Create a new branch, and update version files in the master branch.
 
-    mvn release:branch \
-        -DbranchName=${BRANCH_NAME} \
-        -DdevelopmentVersion=${DEVELOPMENT_VERSION}
+    git branch ${BRANCH}
 
-However, if you are doing an incremental/hotfix release, please run the following command after checking out the release tag of the release being patched.
+    # Now change the version in existing gradle files, and Python files
+    sed -i -e "s/'${RELEASE}'/'${NEXT_VERSION_IN_BASE_BRANCH}'/g" build_rules.gradle
+    sed -i -e "s/${RELEASE}/${NEXT_VERSION_IN_BASE_BRANCH}/g" gradle.properties
+    sed -i -e "s/${RELEASE}/${NEXT_VERSION_IN_BASE_BRANCH}/g" sdks/python/apache_beam/version.py
 
-    mvn release:branch \
-        -DbranchName=${BRANCH_NAME} \
-        -DupdateWorkingCopyVersions=false \
-        -DupdateBranchVersions=true \
-        -DreleaseVersion="${VERSION}-SNAPSHOT"
+    # Save changes in master branch
+    git add gradle.properties build_rules.gradle sdks/python/apache_beam/version.py
+    git commit -m "Moving to ${NEXT_VERSION_IN_BASE_BRANCH}-SNAPSHOT on master branch."
 
 Check out the release branch.
 
-    git checkout ${BRANCH_NAME}
+    git checkout ${BRANCH}
```

Comment:
This step needs to be moved later, after changing versions on master?

Reply:
We do all master update in previous sed scripts. The rest of instructions all happen in release branch.